pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -796,6 +796,10 @@ func buildHTTPConnectionManager(env *model.Environment, node *model.Proxy, httpO
 		// Allow websocket upgrades
 		websocketUpgrade := &http_conn.HttpConnectionManager_UpgradeConfig{UpgradeType: "websocket"}
 		connectionManager.UpgradeConfigs = []*http_conn.HttpConnectionManager_UpgradeConfig{websocketUpgrade}
+		// Setting IdleTimeout to 0 seems to break most tests, causing
+		// envoy to disconnect. See PR comments.
+		// notimeout := 0 * time.Second
+		// connectionManager.IdleTimeout = &notimeout
```

Comment:
But this is the crux of the fix (well for the older envoy). going to try with newer envoy (which requires setting stream_idle_timeout instead)